### ISSUE TITLE
SSL error when sending push notifications

### ISSUE THREAD

**Issue by faaaaabi** [matrix-org/synapse#6650]
I recently installed synapse via docker with following [version](https://hub.docker.com/layers/matrixdotorg/synapse/latest/images/sha256-89029c49844b229fcda1ecf360a21152e9d220c5da7be0218be34b7291066ca3) of the image. Everything works as expected despite the push notifications. I get following error when synapse tries to send push notifications

```
2020-01-07 09:25:07,166 - synapse.push.httppusher - 195 - INFO - httppush.process-275 - Processing 1 unprocessed push actions for @user:domain.tld/im.vector.app.android/mOKyfuLBPaTywZdw0cDbv4E9PRBq9w6GcWEbeHfzIjEQbQHxH0PzCuy09tWyQ17HAXlbuZRCAVjYLAaUMEH19:t3m41OYg9PIjZE8lIELavRjkS-_HPZ-m4RGdJOQA1G0OjIT4tymaodZC8h0GdyV_OQ starting at stream_ordering 58
2020-01-07 09:25:07,167 - synapse.http.client - 284 - INFO - httppush.process-275 - Sending request POST https://matrix.org/_matrix/push/v1/notify
2020-01-07 09:25:07,241 - synapse.http.client - 331 - INFO - httppush.process-275 - Error sending request to  POST https://matrix.org/_matrix/push/v1/notify: RequestTransmissionFailed [<twisted.python.failure.Failure OpenSSL.SSL.Error: [('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')]>]
2020-01-07 09:25:07,241 - synapse.push.httppusher - 387 - WARNING - httppush.process-275 - Failed to push event $E_xxlmrrFVCYfDNdogOnatfDBxcejOyggM93H1mZshn to @user:domain.tld/im.vector.app.android/mOKyfuLBPaTywZdw0cDbv4E9PRBq9w6GcWEbeHfzIjEQbQHxH0PzCuy09tWyQ17HAXlbuZRCAVjYLAaUMEH19:t3m41OYg9PIjZE8lIELavRjkS-_HPZ-m4RGdJOQA1G0OjIT4tymaodZC8h0GdyV_OQ: <class 'twisted.web._newclient.RequestTransmissionFailed'> [<twisted.python.failure.Failure OpenSSL.SSL.Error: [('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')]>]
2020-01-07 09:25:07,241 - synapse.push.httppusher - 272 - INFO - httppush.process-275 - Push failed: delaying for 64s
```
(I shuffeled any ID and changed user names and domains)

I thought that the reason could be that a root cert is not installed on the host system but as I run synapse in a docker container, this should all be installed within the docker image.

Am I missing something?

**Comment by richvdh** [comment]
fixed by #6656 I believe

**Comment by Sr-Preet** [comment]
I'm getting the same error while trying to log in.
Can you please help me with this.
